### PR TITLE
Uploads `terraform plan` summary instead of complete plan output

### PR DESCRIPTION
**Pull request by alex-goncharov**
- adds separate parameter for the plan output: planOutputFilePath to be used lated with `terraform show` command
- when publish results is requested, run terraform show command with planOutputFilePath as the input file.
- add ability to switch off stdout buffer transforts (not sure why are they there in the first place).
- replace content of the attachment with `terraform show` output

ToDo

- [x] planOutputFilePath and publishPlanResults verification
- [] Fix tests with valid `terraform plan` and `terraform display [-json]` outputs

### CODE REVIEW

**Comment by alex-goncharov** [comment]
@charleszipp @favoretti 

I would like to replace the complete `terraform plan` output with only the summary of the plan, pretty much like it was intended in my original draft. The main reason we have started this is to be able to quickly glance at the proposed plan instead of scrolling thousands of lines to get to the bottom just to see that this one parameter of this single resource is planned to be changed the way it should. 

Check if the approach I've taken is something that can go forward, considering the fact that there will be another `terraform show` run with `-json` output, to upload plan summary.

File: tasks/terraform-cli/src/commands/tf-plan.ts [hunk 1]
```diff
@@ -2,23 +2,35 @@ import { CommandResponse, ICommand } from ".";
...
 export const publishedPlanAttachmentType = "terraform-plan-results";
 
+const publishPlanIsNotConfigured = `
+    The publish plan feature is not configured or configured incorrectly.
+    For the feature to work the following paramters of the "plan" command
+    must be set to the correct values:
+    
+    planOutputFilePath: path to the file where terraform plan will be output
+                        by the "terraform plan" command.
+    planOutputFilePath: non-empty string that identifies this plan. This has
```
Comment (favoretti) [inline tasks/terraform-cli/src/commands/tf-plan.ts:18]: This should be `publishPlanResults` I assume?

**Review by favoretti — CHANGES_REQUESTED** [review]
The reason tests are failing is because you haven't added your new parameter to the pipelines configuration that are running the tests, hence the output of the plan differs from what you'd expect since it prints your error message instead of a plan.

File: tasks/terraform-cli/src/commands/tf-plan.ts (hunk 1 above)
Comment (alex-goncharov) [reply tasks/terraform-cli/src/commands/tf-plan.ts:18]: both fixed, 3 tests remaining.

Comment (alex-goncharov) [reply tasks/terraform-cli/src/commands/tf-plan.ts]: it must be , indeed

**Review by charleszipp — CHANGES_REQUESTED** [review]
Appreciate the contribution as always @alex-goncharov. I was hoping to add the summary once we got through the initial release. There was quite a bit of work just to support testing the UI extension so wanted to focus on that first.

As for the approach in this PR I would suggest rethinking a few things. 
1) We need to maintain existing functionality. Since the plan detail is already showing we need to make sure that continues to show since its already been released in the wild.
2) The extra plan out configuration: I dont believe its going to be intuitive for users to know that this is required for publishing plan results to work. If we absolutely need the plan output then I would suggest trying to detect it from command options first. If it wasnt provided, then the task can just generate one on behalf of the user. This way the user doesnt have to update their pipeline to start saving plan output if they didnt need it in the first place.
3) Do we need to run show to get the summary? If we could avoid the need to run show, I think much of this would be simplified. I was hoping that we might be able to parse the plan detail to extract the summary information. Terraform always has a summary line at the end that reports how many resources will be created, destroyed, and updated. Can we regex this and use it for our summary?

**Comment by favoretti** [comment]
@charleszipp By plan detail we actually mean not showing the refresh part of the plan. The actual details are the ones that are interesting, but we have plans with 3000+ resources in them and scrolling through refresh phase of it is just useless, hence the initial idea of this tab - show only what's changing, not state refreshes.

**Comment by charleszipp** [comment]
> @charleszipp By plan detail we actually mean not showing the refresh part of the plan. The actual details are the ones that are interesting, but we have plans with 3000+ resources in them and scrolling through refresh phase of it is just useless, hence the initial idea of this tab - show only what's changing, not state refreshes.

Thanks @favoretti that helps. To double check my understanding, the reason for using `show` is that it will produce the change summary without all the refresh logs correct? If so, lets modify this to do what I suggested in the bullet 2. Check the command options for the existence of `-out`. If one exists, use that when running show. Otherwise, append on behalf of the user when running plan. In the case were we provide it on their behalf, make sure the plan output is saved to agent temp directory so its deleted after task is finished. That should remove the need for extra configuration.

**Comment by favoretti** [comment]
@charleszipp Yes sir, you are completely correct. I'll leave the implementation suggestion up to @alex-goncharov here, since it's his code, unless he wants me to jump in.

**Comment by favoretti** [comment]
#249 will take care of this, closing.

**Pull request closed without merging**